File: app/src/main/java/com/hlabexamples/fabrevealmenu/DemoCodeFragment.java
package com.hlabexamples.fabrevealmenu;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import androidx.core.content.res.ResourcesCompat;
import androidx.appcompat.content.res.AppCompatResources;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.CheckBox;
import android.widget.Spinner;
import android.widget.Toast;

import com.hlab.fabrevealmenu.helper.RevealDirection;
import com.hlab.fabrevealmenu.helper.OnFABMenuSelectedListener;
import com.hlab.fabrevealmenu.model.FABMenuItem;
import com.hlab.fabrevealmenu.view.FABRevealMenu;

import java.util.ArrayList;
import java.util.Objects;

public class DemoCodeFragment extends BaseFragment implements OnFABMenuSelectedListener {

    private ArrayList<FABMenuItem> items;
    private String[] mDirectionStrings = {"LEFT", "UP"};
    private RevealDirection currentRevealDirection = RevealDirection.LEFT;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_code, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        initItems(false);

        final FABRevealMenu fabMenu = initFabMenu(view);

        initListeners(view, fabMenu);
    }

    private FABRevealMenu initFabMenu(@NonNull View view) {
        FloatingActionButton fab = view.findViewById(R.id.fab);
        final FABRevealMenu fabMenu = view.findViewById(R.id.fabMenu);

        try {
            if (fab != null && fabMenu != null) {
                //attach menu to fab
                setFabMenu(fabMenu);
                //set menu items from arrylist
                fabMenu.setMenuItems(items);
                //attach menu to fab
                fabMenu.bindAnchorView(fab);
                //set menu item selection
                fabMenu.setOnFABMenuSelectedListener(this);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return fabMenu;
    }

    private void initListeners(@NonNull View view, FABRevealMenu fabMenu) {
        CheckBox cbTitle = view.findViewById(R.id.chTitle);
        cbTitle.setOnCheckedChangeListener((compoundButton, b) -> {
            if (fabMenu != null) {
                fabMenu.setTitleVisible(b);
            }
        });
        CheckBox cbShowOverlay = view.findViewById(R.id.chOverlay);
        cbShowOverlay.setOnCheckedChangeListener((compoundButton, b) -> {
            if (fabMenu != null) {
                fabMenu.setShowOverlay(b);
            }
        });
        CheckBox cbDouble = view.findViewById(R.id.chDouble);
        cbDouble.setOnCheckedChangeListener((compoundButton, b) -> {
            if (fabMenu != null) {
                initItems(b);
                fabMenu.setMenuItems(items);
            }
        });
        CheckBox cbFont = view.findViewById(R.id.chFont);
        cbFont.setOnCheckedChangeListener((compoundButton, b) -> {
            if (fabMenu != null) {
                //set custom font typeface
                fabMenu.setMenuTitleTypeface(ResourcesCompat.getFont(requireActivity(), R.font.quicksand));
            }
        });
        CheckBox chSmall = view.findViewById(R.id.chSmall);
        chSmall.setOnCheckedChangeListener((compoundButton, isSmaller) -> {
            if (fabMenu != null) {
                if (isSmaller)
                    fabMenu.setSmallerMenu();
                else
                    fabMenu.setNormalMenu();
            }
        });
        Spinner spDirections = view.findViewById(R.id.spDirection);
        spDirections.setAdapter(new ArrayAdapter<>(requireActivity(), android.R.layout.simple_spinner_dropdown_item, mDirectionStrings));
        spDirections.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
                if (fabMenu != null) {
                    if (position == 0 && currentRevealDirection != RevealDirection.LEFT) {
                        currentRevealDirection = RevealDirection.LEFT;
                        fabMenu.setMenuDirection(RevealDirection.LEFT);
                    } else if (position == 1 && currentRevealDirection != RevealDirection.UP) {
                        currentRevealDirection = RevealDirection.UP;
                        fabMenu.setMenuDirection(RevealDirection.UP);
                    }
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {
                if (fabMenu != null) {
                    fabMenu.setMenuDirection(RevealDirection.LEFT);
                }
            }
        });
    }

    private void initItems(boolean toShowDoubleItems) {
        items = new ArrayList<>();
        items.add(new FABMenuItem("Attachments", AppCompatResources.getDrawable(requireActivity(), R.drawable.ic_attachment)));
        items.add(new FABMenuItem("Images", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_image)));
        items.add(new FABMenuItem("Places", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_place)));
        items.add(new FABMenuItem("Emoticons", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_emoticon)));
        if (toShowDoubleItems) {
            items.add(new FABMenuItem("Attachments", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_attachment)));
            items.add(new FABMenuItem("Images", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_image)));
            items.add(new FABMenuItem("Places", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_place)));
            items.add(new FABMenuItem("Emoticons", AppCompatResources.getDrawable(getActivity(), R.drawable.ic_emoticon)));
        }
    }

    @Override
    public void onMenuItemSelected(View view, int id) {
        if (id >= 0 && items != null && items.size() > id) {
            Toast.makeText(getActivity(), items.get(id).getTitle() + "Clicked", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/hlabexamples/fabrevealmenu/DemoCodeFragment.kt
package com.hlabexamples.fabrevealmenu

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.content.res.AppCompatResources
import androidx.core.content.res.ResourcesCompat
import androidx.fragment.app.Fragment
import com.hlab.fabrevealmenu.enums.Direction
import com.hlab.fabrevealmenu.listeners.OnFABMenuSelectedListener
import com.hlab.fabrevealmenu.model.FABMenuItem
import kotlinx.android.synthetic.main.fragment_code.*

class DemoCodeFragment : Fragment(), OnFABMenuSelectedListener {

    private var items = arrayListOf<FABMenuItem>()
    private val mDirectionStrings = arrayOf("LEFT", "UP")
    private var currentDirection = Direction.LEFT

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_code, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        initItems(false)

        try {
            if (fab != null && fabMenu != null) {
                //attach menu to fab
                //set menu items from arrylist
                fabMenu.setMenuItems(items)
                //attach menu to fab
                fabMenu.bindAnchorView(fab)
                //set menu item selection
                fabMenu.setOnFABMenuSelectedListener(this)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

        val cbTitle = view.findViewById<CheckBox>(R.id.chTitle)
        cbTitle.setOnCheckedChangeListener { _, isChecked ->
            fabMenu?.setTitleVisible(isChecked)
        }
        val cbShowOverlay = view.findViewById<CheckBox>(R.id.chOverlay)
        cbShowOverlay.setOnCheckedChangeListener { _, isChecked ->
            if (fabMenu != null) {
                fabMenu.isShowOverlay = isChecked
            }
        }
        val cbDouble = view.findViewById<CheckBox>(R.id.chDouble)
        cbDouble.setOnCheckedChangeListener { buttonView, isChecked ->
            if (fabMenu != null) {
                initItems(isChecked)
                fabMenu.setMenuItems(items)
            }
        }
        val cbFont = view.findViewById<CheckBox>(R.id.chFont)
        cbFont.setOnCheckedChangeListener { buttonView, isChecked ->
            fabMenu?.setMenuTitleTypeface(ResourcesCompat.getFont(activity!!, R.font.quicksand))
        }
        val chSmall = view.findViewById<CheckBox>(R.id.chSmall)
        chSmall.setOnCheckedChangeListener { buttonView, isChecked ->
            if (fabMenu != null) {
                if (isChecked)
                    fabMenu.setSmallerMenu()
                else
                    fabMenu.setNormalMenu()
            }
        }
        val chAnimate = view.findViewById<CheckBox>(R.id.chAnimate)
        chAnimate.setOnCheckedChangeListener { _, isChecked ->
            fabMenu?.enableItemAnimation(isChecked)
        }

        val spDirections = view.findViewById<Spinner>(R.id.spDirection)
        spDirections.adapter = ArrayAdapter(activity!!, android.R.layout.simple_spinner_dropdown_item, mDirectionStrings)
        spDirections.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(adapterView: AdapterView<*>, view: View, position: Int, l: Long) {
                if (fabMenu != null) {
                    if (position == 0 && currentDirection != Direction.LEFT) {
                        currentDirection = Direction.LEFT
                        fabMenu.menuDirection = Direction.LEFT
                    } else if (position == 1 && currentDirection != Direction.UP) {
                        currentDirection = Direction.UP
                        fabMenu.menuDirection = Direction.UP
                    }
                }
            }

            override fun onNothingSelected(adapterView: AdapterView<*>) {
                if (fabMenu != null) {
                    fabMenu.menuDirection = Direction.LEFT
                }
            }
        }

        fabMenu.setOverlayBackground(R.color.colorAccent)
        fabMenu.setMenuBackground(R.color.colorWhite)
    }

    private fun initItems(toShowDoubleItems: Boolean) {
        items.clear()
        items.add(FABMenuItem("Attachments", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_attachment)))
        items.add(FABMenuItem("Images", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_image)))
        items.add(FABMenuItem("Places", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_place)))
        items.add(FABMenuItem("Emoticons", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_emoticon)))
        if (toShowDoubleItems) {
            items.add(FABMenuItem("Attachments", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_attachment)))
            items.add(FABMenuItem("Images", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_image)))
            items.add(FABMenuItem("Places", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_place)))
            items.add(FABMenuItem("Emoticons", iconDrawable = AppCompatResources.getDrawable(activity!!, R.drawable.ic_emoticon)))
        }
    }

    override fun onMenuItemSelected(view: View, id: Int) {
        if (id >= 0 && items.size > id) {
            Toast.makeText(activity, items[id].title + "Clicked", Toast.LENGTH_SHORT).show()
        }
    }
}